internal/dataplane/kong_client.go

```diff
@@ -740,6 +740,12 @@ func (c *KongClient) maybeSendOutToKonnectClient(
 		return nil
 	}
 
+	// In case users have many consumers, konnect sync can be very slow and cause dataplane sync issues.
+	// For this reason, if the --disable-consumers-sync flag is set, we do not send consumers to Konnect.
+	if konnectClient.ConsumersSyncDisabled() {
+		s.Consumers = nil
```

Comment:
Would it be better to set an option in `ToDeckContent` to control the generated config? modifying a shared pointer is quite dangerous in concurrent sending (would be required in #6188.)

Reply:
When we implement the concurrent sending, we'll have to duplicate the payload in my opinion, as the konnect kongstate can be different from the Gateway one. For example, if this flag is enabled, konnect's payload won't have any consumer, while the gateway's will.